src/MiddleMail/MiddleMailService.cs

```diff
@@ -43,7 +62,30 @@ protected async override Task ExecuteAsync(CancellationToken cancellationToken)
 			}
 		}
 
+		/// <summary>
+		/// If rate limiting is configured, acquires a lease,
+		/// awaiting the RateLimitDelay at each failure to acquire
+		/// until the lease is successfully acquired.
+		/// This means that this function will not return
+		/// until the number of emails to be sent is back below the rate limit.
+		/// If rate limiting is not configured, returns null.
+		/// (A using statement that uses null suceeds without any problem.)
+		/// </summary>
+		/// <returns></returns>
+		private async Task<RateLimitLease?> acquireLease() {
+			if (rateLimiter == default) return null;
+
+			while (true) {
+				RateLimitLease lease = await rateLimiter.AcquireAsync();
+				if (lease.IsAcquired) return lease;
+
+				await Task.Delay(RateLimitDelay);
+			}
+		}
+
 		private async Task processAsync(EmailMessage emailMessage ) {
+			using RateLimitLease? lease = await acquireLease();
```

Comment:
Because EasyNetQ only considers the message delivery "acked" once the delegate returns, it should retry the message once the service is started up again.
https://github.com/EasyNetQ/EasyNetQ/wiki/Subscribe